From 19f02c422e21ece13a2e4072ae47903eb916f7bd Mon Sep 17 00:00:00 2001
From: Lei Zhang <thestig@chromium.org>
Date: Wed, 28 Jul 2021 18:19:39 +0000
Subject: [PATCH] Add missing <string.h> includes for memcpy() and friends.

Found by GCC.

Change-Id: I17db736d4f9bead5f25977823e476bf384ff797f
Reviewed-on: https://pdfium-review.googlesource.com/c/pdfium/+/83554
Commit-Queue: Lei Zhang <thestig@chromium.org>
Commit-Queue: Tom Sepez <tsepez@chromium.org>
Auto-Submit: Lei Zhang <thestig@chromium.org>
Reviewed-by: Tom Sepez <tsepez@chromium.org>
---

diff --git a/third_party/pdfium/core/fdrm/fx_crypt.cpp b/third_party/pdfium/core/fdrm/fx_crypt.cpp
index 5208dcf..ce96f9b 100644
--- a/third_party/pdfium/core/fdrm/fx_crypt.cpp
+++ b/third_party/pdfium/core/fdrm/fx_crypt.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fdrm/fx_crypt.h"
 
+#include <string.h>
+
 #include <utility>
 
 #define GET_UINT32(n, b, i)                            \
diff --git a/third_party/pdfium/core/fdrm/fx_crypt_aes.cpp b/third_party/pdfium/core/fdrm/fx_crypt_aes.cpp
index c5718f9..f217022 100644
--- a/third_party/pdfium/core/fdrm/fx_crypt_aes.cpp
+++ b/third_party/pdfium/core/fdrm/fx_crypt_aes.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fdrm/fx_crypt.h"
 
+#include <string.h>
+
 #include "third_party/base/check.h"
 #include "third_party/base/check_op.h"
 
diff --git a/third_party/pdfium/core/fdrm/fx_crypt_sha.cpp b/third_party/pdfium/core/fdrm/fx_crypt_sha.cpp
index f5fea2c..5e62eb7 100644
--- a/third_party/pdfium/core/fdrm/fx_crypt_sha.cpp
+++ b/third_party/pdfium/core/fdrm/fx_crypt_sha.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fdrm/fx_crypt.h"
 
+#include <string.h>
+
 #define SHA_GET_UINT32(n, b, i)                                         \
   {                                                                     \
     (n) = ((uint32_t)(b)[(i)] << 24) | ((uint32_t)(b)[(i) + 1] << 16) | \
diff --git a/third_party/pdfium/core/fxcodec/flate/flatemodule.cpp b/third_party/pdfium/core/fxcodec/flate/flatemodule.cpp
index 316e02d..b51b073 100644
--- a/third_party/pdfium/core/fxcodec/flate/flatemodule.cpp
+++ b/third_party/pdfium/core/fxcodec/flate/flatemodule.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxcodec/flate/flatemodule.h"
 
+#include <string.h>
+
 #include <algorithm>
 #include <limits>
 #include <memory>
diff --git a/third_party/pdfium/core/fxcodec/jbig2/jbig2_decoder.cpp b/third_party/pdfium/core/fxcodec/jbig2/jbig2_decoder.cpp
index 5896fa8..adbfbf8 100644
--- a/third_party/pdfium/core/fxcodec/jbig2/jbig2_decoder.cpp
+++ b/third_party/pdfium/core/fxcodec/jbig2/jbig2_decoder.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxcodec/jbig2/jbig2_decoder.h"
 
+#include <string.h>
+
 #include "core/fxcodec/jbig2/JBig2_Context.h"
 #include "core/fxcodec/jbig2/JBig2_DocumentContext.h"
 
diff --git a/third_party/pdfium/core/fxcodec/png/png_decoder.cpp b/third_party/pdfium/core/fxcodec/png/png_decoder.cpp
index af3fcb9..58cf818 100644
--- a/third_party/pdfium/core/fxcodec/png/png_decoder.cpp
+++ b/third_party/pdfium/core/fxcodec/png/png_decoder.cpp
@@ -6,6 +6,8 @@
 
 #include "core/fxcodec/png/png_decoder.h"
 
+#include <string.h>
+
 #include <algorithm>
 
 #include "core/fxcodec/cfx_codec_memory.h"
